src/resmom/mom_job_func.c

```diff
@@ -314,8 +314,8 @@ int remtree(
 
     while ((pdir = readdir(dir)) != NULL)
       {
-      if ((pdir->d_name[0] == '.') &&
-          ((pdir->d_name[1] == '\0') || (pdir->d_name[1] == '.')))
+      if (pdir->d_name[0] == '.' && pdir->d_name[1] == '\0' ||
+	  pdir->d_name[1] == '.' && pdir->d_name[2] == '\0')
```

Comment:
I don't think this will work correctly as written.  This will skip items named, for example, "a." or "2."  Logic expressions are *not* simply evaluated left-to-right but rather follow rules of precedence, order of operations, and short-circuiting; "&&" does not have the same precedence as "||" has, and multiple OR'd expressions are evaluated unless/until one evaluates to true.  Parentheses are needed to ensure that the entire expression evaluates to false if the first sub-expression is false.  It should be:

`if (pdir->d_name[0] == '.' && (pdir->d_name[1] == '\0' || pdir->d_name[1] == '.' && pdir->d_name[2] == '\0'))`

But I'm a firm believer in fully parenthesizing boolean expressions because it makes the author's intended logic clearer to a future reader, so I'd write it like so:

```
if ((pdir->d_name[0] == '.')
    && ((pdir->d_name[1] == '\0')
         || ((pdir->d_name[1] == '.') && (pdir->d_name[2] == '\0'))))
```